speaker/layer2_controller.go

```diff
@@ -61,10 +60,9 @@ func usableNodes(eps []discovery.EndpointSlice, speakers map[string]bool) []stri
 				continue
 			}
 			nodeName := *ep.NodeName
-			if speakers != nil {
-				if hasSpeaker := speakers[nodeName]; !hasSpeaker {
-					continue
-				}
+			//DISCUSS: we can keep if speaker !=nil to be safe but that is not a valid code path
```

Comment:
what I was suggesting is to have this function accept SpeakerListInfo instead of the map, and to rely on the Disabled field instead of having a nil map.

Reply:
I might be confused

But this function receives a map which is never nil
1. if memberList disabled, the  arg into that function is the full list of nodes 
2. if memberList enabled, the arg into that function is a map which can be empty by design but not nil

Reply:
now I got the logic of your implementation, it's fine as it is. Thanks.